### PR TITLE
fix/PN-9620: fix spacing in modal PF/PA/PG for tablet

### PR DESCRIPTION
**Pull request by SarahDonvito**
## List of changes proposed in this pull request
- removed isMobile hook where not needed
- added padding property in DialogTitle, DialogContent and DialogAction with breakpoints
- fixed direction and size of button in DialogAction

## How to test
Login as PF/PG/PA and check in tablet:
- padding of entire modal: 32px
- bottom distance between title and body: 16px
- bottom distance between title actions: 32px
- bottom distance between title elements in body: 16px

### CODE REVIEW

File: packages/pn-commons/src/components/CodeModal/CodeModal.tsx [hunk 1]
```diff
@@ -90,7 +90,9 @@ const CodeModal = memo(
         data-testid="codeDialog"
         disableEscapeKeyDown
       >
-        <DialogTitle id="dialog-title">{title}</DialogTitle>
+        <DialogTitle sx={{ pb: { xs: 2, sm: 2 } }} id="dialog-title">
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/CodeModal/CodeModal.tsx:93]: The sx attribute is useless

File: packages/pn-commons/src/components/PnDialog/PnDialogActions.tsx [hunk 2]
```diff
@@ -7,19 +7,16 @@ import { ReactComponent } from '../../models/PnDialog';
...
   );
 
-  const enrichedButtons = buttons.map((button, index) =>
+  const enrichedButtons = buttons.map((button) =>
     isValidElement(button)
       ? cloneElement(button, {
           ...button.props,
-          fullWidth: isMobile,
-          sx: { marginBottom: isMobile && index > 0 ? 2 : 0, ...button.props.sx },
+          sx: { marginBottom: { xs: 2, sm: 0 }, width: { xs: 1, sm: 'auto' }, ...button.props.sx },
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/PnDialog/PnDialogActions.tsx:19]: Restore the condition index > 0 that was on previous version

File: packages/pn-commons/src/components/PnDialog/PnDialogActions.tsx [hunk 3]
```diff
@@ -30,10 +27,10 @@ const PnDialogActions: React.FC<DialogActionsProps> = (props) => {
       disableSpacing={isMobile}
       {...props}
       sx={{
-        flexDirection: isMobile ? 'column-reverse' : 'row',
-        p: paddingSize,
-        pt: 0,
-        gap: gapSize,
+        flexDirection: { xs: 'column-reverse', sm: 'row' },
+        p: { xs: 3, sm: 4 },
+        pt: { xs: 0, sm: 0 },
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/PnDialog/PnDialogActions.tsx:32]: put 0

File: packages/pn-commons/src/components/PnDialog/__test__/PnDialog.test.tsx [hunk 4]
```diff
@@ -9,8 +9,10 @@ describe('PnDialog Component', () => {
     window.matchMedia = createMatchMedia(2000);
     const { queryByTestId } = render(
       <PnDialog open>
-        <DialogTitle data-testid="dialog-title">Title</DialogTitle>
-        <PnDialogContent>
+        <DialogTitle sx={{ p: 4, pb: 2 }} data-testid="dialog-title">
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/PnDialog/__test__/PnDialog.test.tsx:12]: The sx attribute is useless

File: packages/pn-commons/src/components/PnDialog/__test__/PnDialog.test.tsx [hunk 5]
```diff
@@ -9,8 +9,10 @@ describe('PnDialog Component', () => {
...
+          Title
+        </DialogTitle>
+        <PnDialogContent sx={{ p: 4 }}>
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/PnDialog/__test__/PnDialog.test.tsx:15]: The sx attribute is useless

File: packages/pn-commons/src/components/PnDialog/__test__/PnDialog.test.tsx [hunk 6]
```diff
@@ -29,8 +31,10 @@ describe('PnDialog Component', () => {
     window.matchMedia = createMatchMedia(800);
     const { queryByTestId } = render(
       <PnDialog open>
-        <DialogTitle data-testid="dialog-title">Title</DialogTitle>
-        <PnDialogContent>
+        <DialogTitle sx={{ p: 3, pb: 2 }} data-testid="dialog-title">
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/PnDialog/__test__/PnDialog.test.tsx:34]: The sx attribute is useless

File: packages/pn-commons/src/components/PnDialog/__test__/PnDialog.test.tsx [hunk 7]
```diff
@@ -29,8 +31,10 @@ describe('PnDialog Component', () => {
...
+          Title
+        </DialogTitle>
+        <PnDialogContent sx={{ p: 3 }}>
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/PnDialog/__test__/PnDialog.test.tsx:37]: The sx attribute is useless

File: packages/pn-commons/src/components/PnDialog/__test__/PnDialogActions.test.tsx [hunk 8]
```diff
@@ -7,7 +7,14 @@ describe('PnDialogActions Component', () => {
   it('renders component - desktop', () => {
     window.matchMedia = createMatchMedia(2000);
     const { queryByTestId, queryAllByTestId } = render(
-      <PnDialogActions>
+      <PnDialogActions
+        sx={{
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/PnDialog/__test__/PnDialogActions.test.tsx:11]: The sx attribute is useless

File: packages/pn-commons/src/components/PnDialog/__test__/PnDialogActions.test.tsx [hunk 9]
```diff
@@ -28,9 +35,18 @@ describe('PnDialogActions Component', () => {
   it('renders component - mobile', () => {
     window.matchMedia = createMatchMedia(800);
     const { queryByTestId, queryAllByTestId } = render(
-      <PnDialogActions>
+      <PnDialogActions
+        sx={{
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/PnDialog/__test__/PnDialogActions.test.tsx:39]: The sx attribute is useless

File: packages/pn-commons/src/components/PnDialog/__test__/PnDialogActions.test.tsx [hunk 10]
```diff
@@ -28,9 +35,18 @@ describe('PnDialogActions Component', () => {
...
+          flexDirection: 'column-reverse',
+          p: 3,
+          pt: 0,
+          gap: 0,
+        }}
+      >
         <Button data-testid="button">Test confirm button</Button>
-        <Button data-testid="button">Test cancel button</Button>
+        <Button sx={{ marginBottom: 2, width: 1 }} data-testid="button">
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/PnDialog/__test__/PnDialogActions.test.tsx:47]: The sx attribute is useless

File: packages/pn-commons/src/components/Prompt.tsx [hunk 11]
```diff
@@ -40,7 +40,7 @@ const Prompt: React.FC<Props> = ({
   return (
     <>
       <PnDialog onClose={cancelNavigation} open={showPrompt} data-testid="promptDialog">
-        <DialogTitle>{title}</DialogTitle>
+        <DialogTitle sx={{ pb: { xs: 2, sm: 2 } }}>{title}</DialogTitle>
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/Prompt.tsx:43]: The sx attribute is useless

File: packages/pn-commons/src/components/SessionModal.tsx [hunk 12]
```diff
@@ -66,7 +66,7 @@ const SessionModal: React.FC<Props> = ({
       aria-labelledby="session-dialog-title"
       data-testid="session-modal"
     >
-      <DialogTitle sx={{ textAlign: 'center' }}>{title}</DialogTitle>
+      <DialogTitle sx={{ textAlign: 'center', pb: { xs: 2, sm: 2 } }}>{title}</DialogTitle>
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/SessionModal.tsx:69]: The pb property is useless

File: packages/pn-pa-webapp/src/components/ApiKeys/ApiKeyModal.tsx [hunk 13]
```diff
@@ -25,7 +25,7 @@ const ApiKeyModal = ({
   actionHandler,
 }: ApiKeyModalProps) => (
   <PnDialog open onClose={closeModalHandler}>
-    {title && <DialogTitle>{title}</DialogTitle>}
+    {title && <DialogTitle sx={{ pb: { xs: 2, sm: 2 } }}>{title}</DialogTitle>}
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/components/ApiKeys/ApiKeyModal.tsx:28]: The sx attribute is useless

File: packages/pn-pa-webapp/src/components/Notifications/ConfirmCancellationDialog.tsx [hunk 14]
```diff
@@ -35,7 +35,9 @@ const ConfirmCancellationDialog: React.FC<Props> = ({ showModal, onClose, onConf
 
   return (
     <PnDialog open={showModal} data-testid="cancel-notification-modal">
-      <DialogTitle id="dialog-title">{t('detail.cancel-notification-modal.title')}</DialogTitle>
+      <DialogTitle sx={{ pb: { xs: 2, sm: 2 } }} id="dialog-title">
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/components/Notifications/ConfirmCancellationDialog.tsx:38]: The sx attribute is useless

File: packages/pn-pa-webapp/src/components/Notifications/NotificationPaymentF24.tsx [hunk 15]
```diff
@@ -28,7 +28,9 @@ const NotificationPaymentF24: React.FC<Props> = ({ iun, payments }) => {
         aria-labelledby="dialog-title"
         maxWidth="lg"
       >
-        <DialogTitle id="dialog-title">{t('payment.f24-dialog-title', { iun })}</DialogTitle>
+        <DialogTitle sx={{ pb: { xs: 2, sm: 2 } }} id="dialog-title">
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/components/Notifications/NotificationPaymentF24.tsx:31]: The sx attribute is useless

File: packages/pn-pa-webapp/src/components/Notifications/NotificationRecipientsDetail.tsx [hunk 16]
```diff
@@ -28,7 +29,9 @@ const NotificationRecipientsDetail: React.FC<Props> = ({ recipients, iun }) => {
         aria-labelledby="dialog-title"
         maxWidth="lg"
       >
-        <DialogTitle id="dialog-title">{t('detail.recipients-dialog-title', { iun })}</DialogTitle>
+        <DialogTitle sx={{ pb: { xs: 2, sm: 2 } }} id="dialog-title">
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/components/Notifications/NotificationRecipientsDetail.tsx:32]: The sx attribute is useless

File: packages/pn-personafisica-webapp/src/components/Contacts/CancelVerificationModal.tsx [hunk 17]
```diff
@@ -24,7 +24,7 @@ const CancelVerificationModal = ({ open, handleClose }: Props) => {
 
   return (
     <PnDialog open={open} onClose={handleClose} data-testid="cancelVerificationModal">
-      <DialogTitle id="dialog-title">
+      <DialogTitle sx={{ pb: { xs: 2, sm: 2 } }} id="dialog-title">
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/CancelVerificationModal.tsx:27]: The sx attribute is useless

File: packages/pn-personafisica-webapp/src/components/Contacts/DigitalContactElem.tsx [hunk 18]
```diff
@@ -85,7 +85,9 @@ const DeleteDialog: React.FC<DialogProps> = ({
       aria-labelledby="dialog-title"
       aria-describedby="dialog-description"
     >
-      <DialogTitle id="dialog-title">{removeModalTitle}</DialogTitle>
+      <DialogTitle sx={{ pb: { xs: 2, sm: 2 } }} id="dialog-title">
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/DigitalContactElem.tsx:88]: The sx attribute is useless

File: packages/pn-personafisica-webapp/src/components/Contacts/DigitalContactsCodeVerification.context.tsx [hunk 19]
```diff
@@ -379,7 +379,7 @@ const DigitalContactsCodeVerificationProvider: FC<{ children?: ReactNode }> = ({
         aria-describedby="dialog-description"
         data-testid="duplicateDialog"
       >
-        <DialogTitle id="dialog-title">
+        <DialogTitle sx={{ pb: { xs: 2, sm: 2 } }} id="dialog-title">
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/DigitalContactsCodeVerification.context.tsx:382]: The sx attribute is useless

File: packages/pn-personafisica-webapp/src/components/Contacts/DigitalContactsCodeVerification.context.tsx [hunk 20]
```diff
@@ -401,7 +401,7 @@ const DigitalContactsCodeVerificationProvider: FC<{ children?: ReactNode }> = ({
         data-testid="validationDialog"
         aria-labelledby="dialog-title"
       >
-        <DialogTitle id="dialog-title">
+        <DialogTitle sx={{ pb: { xs: 2, sm: 2 } }} id="dialog-title">
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/DigitalContactsCodeVerification.context.tsx:404]: The sx attribute is useless

File: packages/pn-personagiuridica-webapp/src/components/Contacts/CancelVerificationModal.tsx [hunk 21]
```diff
@@ -24,7 +24,7 @@ const CancelVerificationModal = ({ open, handleClose }: Props) => {
 
   return (
     <PnDialog open={open} onClose={handleClose} data-testid="cancelVerificationModal">
-      <DialogTitle id="dialog-title">
+      <DialogTitle sx={{ pb: { xs: 2, sm: 2 } }} id="dialog-title">
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Contacts/CancelVerificationModal.tsx:27]: The sx attribute is useless

File: packages/pn-personagiuridica-webapp/src/components/Contacts/DigitalContactElem.tsx [hunk 22]
```diff
@@ -87,7 +87,9 @@ const DeleteDialog: React.FC<DialogProps> = ({
       aria-labelledby="dialog-title"
       aria-describedby="dialog-description"
     >
-      <DialogTitle id="dialog-title">{removeModalTitle}</DialogTitle>
+      <DialogTitle sx={{ pb: { xs: 2, sm: 2 } }} id="dialog-title">
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Contacts/DigitalContactElem.tsx:90]: The sx attribute is useless

File: packages/pn-personagiuridica-webapp/src/components/Contacts/DigitalContactsCodeVerification.context.tsx [hunk 23]
```diff
@@ -326,7 +326,7 @@ const DigitalContactsCodeVerificationProvider: FC<{ children?: ReactNode }> = ({
         aria-describedby="dialog-description"
         data-testid="duplicateDialog"
       >
-        <DialogTitle id="dialog-title">
+        <DialogTitle sx={{ pb: { xs: 2, sm: 2 } }} id="dialog-title">
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Contacts/DigitalContactsCodeVerification.context.tsx:329]: The sx attribute is useless

File: packages/pn-personagiuridica-webapp/src/components/Contacts/DigitalContactsCodeVerification.context.tsx [hunk 24]
```diff
@@ -344,7 +344,7 @@ const DigitalContactsCodeVerificationProvider: FC<{ children?: ReactNode }> = ({
         </PnDialogActions>
       </PnDialog>
       <PnDialog open={pecValidationOpen} data-testid="validationDialog">
-        <DialogTitle id="dialog-title">
+        <DialogTitle sx={{ pb: { xs: 2, sm: 2 } }} id="dialog-title">
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Contacts/DigitalContactsCodeVerification.context.tsx:347]: The sx attribute is useless

File: packages/pn-personagiuridica-webapp/src/components/Deleghe/AcceptDelegationModal.tsx [hunk 25]
```diff
@@ -147,7 +147,7 @@ const AcceptDelegationModal: React.FC<Props> = ({
       aria-describedby="dialog-description"
       data-testid="groupDialog"
     >
-      <DialogTitle id="dialog-title">
+      <DialogTitle sx={{ pb: { xs: 2, sm: 2 } }} id="dialog-title">
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Deleghe/AcceptDelegationModal.tsx:150]: The sx attribute is useless

File: packages/pn-personagiuridica-webapp/src/components/Deleghe/ConfirmationModal.tsx [hunk 26]
```diff
@@ -31,7 +31,9 @@ export default function ConfirmationModal({
       aria-labelledby="responsive-dialog-title"
       data-testid="confirmationDialog"
     >
-      <DialogTitle id="dialog-title">{title}</DialogTitle>
+      <DialogTitle sx={{ pb: { xs: 2, sm: 2 } }} id="dialog-title">
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Deleghe/ConfirmationModal.tsx:34]: The sx attribute is useless

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-commons/src/components/CodeModal/__test__/CodeInput.test.tsx [hunk 27]
```diff
@@ -105,7 +105,7 @@ describe('CodeInput Component', () => {
     act(() => (codeInputs[2] as HTMLInputElement).focus());
     await userEvent.keyboard('{Backspace}');
     await waitFor(() => {
-      expect(codeInputs[2]).toHaveValue('');
+      expect(codeInputs[2]).toBeEmptyDOMElement();
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/CodeModal/__test__/CodeInput.test.tsx:108]: the toBeEmptyDOMElement is to check if an element is empty (i.e. doesn't have any element inside). So an input always will be empty.
Restore the toHaveValue('')

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-commons/src/components/CodeModal/__test__/CodeInput.test.tsx (hunk 27 above)
Comment (SarahDonvito) [reply packages/pn-commons/src/components/CodeModal/__test__/CodeInput.test.tsx]: Okay about toBeEmptyDOMElement, btw using toHaveValue('') sometimes the test fails

**Review by AndreaCimini90 — APPROVED** [review]

**Pull request merged**